x-pack/plugins/ml/public/application/jobs/jobs_list/components/datafeed_modal/datafeed_modal.tsx

```diff
@@ -0,0 +1,281 @@
+          isSelected={tab.id === selectedTabId}
+          disabled={tab.disabled}
+          key={index}
+        >
+          {tab.name}
+        </EuiTab>
+      )),
+    [selectedTabId]
+  );
+
+  const handleChange = (date: moment.Moment) => setEndDate(date);
+
+  const handleEndDateChange = (direction: ChartDirectionType) => {
+    const newEndDate = endDate.clone();
+    const [count, type] = interval.split(' ');
+
+    if (direction === CHART_DIRECTION.FORWARD) {
+      newEndDate.add(Number(count), type);
+    } else {
+      newEndDate.subtract(Number(count), type);
+    }
+    setEndDate(newEndDate);
+  };
+
+  const getChartData = useCallback(async () => {
+    const endTimestamp = moment(endDate).valueOf();
+    const [count, type] = interval.split(' ');
+    const startMoment = endDate.clone().subtract(Number(count), type);
+    const startTimestamp = moment(startMoment).valueOf();
+
+    try {
+      const chartData = await ml.results.getDatafeedResultChartData(
+        jobId,
+        timefield,
+        bucketSpan,
+        startTimestamp,
+        endTimestamp,
+        datafeedConfig
+      );
+
+      setSourceData(chartData.datafeedResults);
+      setBucketData(chartData.bucketResults);
+    } catch (error) {
+      const title = i18n.translate('xpack.ml.jobsList.datafeedModal.errorToastTitle', {
+        defaultMessage: 'Error fetching data',
+      });
+      displayErrorToast(error, title);
+    }
+    setIsLoadingChartData(false);
+  }, [endDate, interval]);
+
+  useEffect(
+    function loadChartData() {
+      setIsLoadingChartData(true);
+      getChartData();
+    },
+    [endDate, interval]
+  );
+
+  return (
+    <EuiModal
+      onClose={onClose.bind(null, false)}
+      className="mlDatafeedModal"
+      style={{ height: '600px', minWidth: '800px' }}
+    >
+      <EuiModalHeader>
+        <EuiFlexGroup alignItems="center" justifyContent="spaceBetween" gutterSize="xl">
+          <EuiFlexItem grow={false}>
+            <FormattedMessage
+              id="xpack.ml.jobsList.datafeedModal.header"
+              defaultMessage="{jobId}"
+              values={{
+                jobId,
+              }}
+            />
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiDatePicker
+              showTimeSelect
+              selected={endDate}
+              onChange={handleChange}
+              popoverPlacement="left-start"
+            />
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiModalHeader>
+
+      <EuiModalBody>
+        <EuiTabs size="s">{renderTabs()}</EuiTabs>
+        <EuiSpacer size="m" />
+        {isLoadingChartData && <EuiLoadingChart size="l" />}
+        {!isLoadingChartData && selectedTabId === TAB_IDS.CHART && (
+          <EuiFlexGroup direction="column">
+            <EuiFlexItem grow={false}>
+              <EuiFlexGroup justifyContent="spaceBetween">
+                <EuiFlexItem grow={false}>
+                  <EuiSelect
```

Comment:
I am not convinced we need this interval dropdown. We could just show the maximum number of points that can be rendered clearly by the chart (480? 960? - probably some multiple of 12). The user can already navigate easily with the back and forward buttons and the time picker control.

The options in the select don't seem to be optimized for longer bucket spans - here for example I have a job with a `12h` bucket span, and it opens with a default of `3h`. Removing the range control would remove the complexity of working out the best range of options to show depending on the bucket span, which could range from seconds to days for example.

![image](https://user-images.githubusercontent.com/7405507/121226656-b56c2800-c882-11eb-8ad1-eb7801733fbf.png)